src/rgw/driver/sfs/sqlite/sqlite_users.cc

```diff
@@ -16,111 +16,117 @@
 #include <filesystem>
 #include <iostream>
 
-#include "users/users_conversions.h"
-
-using namespace sqlite_orm;
+#include "dbapi.h"
+#include "sqlite_query_utils.h"
 
 namespace rgw::sal::sfs::sqlite {
 
 SQLiteUsers::SQLiteUsers(DBConnRef _conn) : conn(_conn) {}
 
 std::optional<DBOPUserInfo> SQLiteUsers::get_user(const std::string& userid
 ) const {
-  auto storage = conn->get_storage();
-  auto user = storage->get_pointer<DBUser>(userid);
-  std::optional<DBOPUserInfo> ret_value;
-  if (user) {
-    ret_value = get_rgw_user(*user);
-  }
-  return ret_value;
+  return GetSQLiteSingleObject<DBOPUserInfo>(
+      conn->get(), "users", "user_id", userid
+  );
 }
 
 std::optional<DBOPUserInfo> SQLiteUsers::get_user_by_email(
     const std::string& email
 ) const {
-  auto users = get_users_by(where(c(&DBUser::user_email) = email));
-  std::optional<DBOPUserInfo> ret_value;
-  if (users.size()) {
-    ret_value = users[0];
-  }
-  return ret_value;
+  return GetSQLiteSingleObject<DBOPUserInfo>(
+      conn->get(), "users", "user_email", email
+  );
 }
 
 std::optional<DBOPUserInfo> SQLiteUsers::get_user_by_access_key(
     const std::string& key
 ) const {
-  auto storage = conn->get_storage();
-  auto user_id = _get_user_id_by_access_key(storage, key);
-  std::optional<DBOPUserInfo> ret_value;
-  if (user_id.has_value()) {
-    auto user = storage->get_pointer<DBUser>(user_id);
-    if (user) {
-      ret_value = get_rgw_user(*user);
-    }
-  }
-  return ret_value;
+  auto user_id = _get_user_id_by_access_key(key);
+  return GetSQLiteSingleObject<DBOPUserInfo>(
+      conn->get(), "users", "user_id", user_id
+  );
 }
 
 std::vector<std::string> SQLiteUsers::get_user_ids() const {
-  auto storage = conn->get_storage();
-  return storage->select(&DBUser::user_id);
+  dbapi::sqlite::database db = conn->get();
+  auto rows = db << R"sql(SELECT user_id FROM users;)sql";
+  std::vector<std::string> ret;
+  for (std::tuple<std::string> row : rows) {
```

Comment:
maybe you can use the `&&` notation here if `row` can be moved and use `std::move(row)` in the subsequent constructor

Reply:
We know in this case that the type returned by the query is `std::tuple<std::string>`. 

I used the && notation in the templated code because we don't know the type at that point.
I've also tested using the deduction form (`auto&&`) and the tuple form and in both cases the library is creating a `row_iterator &` and we can always cast to an rvalue using `std::move`. Anyway, all this will be revisited when the port to sqlite modern cpp is finished.

In this case it's the std::sting inside the tuple the one that we can move. 
I've changed the code to move the string.